### PR TITLE
BK-197: Image mask

### PR DESCRIPTION
**Pull request by thomasbrockmeier-ams**
Adds code to generate masks for panorama images based on the heading of the car. These masks are to be used to obscure the view from the front and back of the car, so only the the right and left side of the image is preserved.

### CODE REVIEW

File: src/api_request.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def get_pano_location(pano_id):
     
     # WGS84 to RD conversion 
     point = ogr.Geometry(ogr.wkbPoint)
-    point.AddPoint(geom[0], geom[1])
+    point.AddPoint(max(geom), min(geom))
```
Comment (thomasbrockmeier-ams) [inline src/api_request.py:26]: This is ugly, but it works because the point has to be in the Netherlands anyway since we're projecting to RD NEW.

**Comment by thomasbrockmeier-ams** [comment]
This should have tests :stuck_out_tongue_winking_eye: but I want to package this repo and extract this functionality to the data pipeline (and have the tests live there). It seems a bit much to add a bunch of testing functionality to this repo, just for a piece of code that will be moved later.

File: src/api_request.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def get_pano_data(pano_id):
         return
     
     # Get panoramic image from API
-    image_url = pano_data['_links']['equirectangular_small']['href']
+    image_url = pano_data['_links']['equirectangular_medium']['href']
```
Comment (epureanudiana) [inline src/api_request.py:53]: What is the difference between this and get_panorama function?

File: src/api_request.py (hunk 1 above)
Comment (epureanudiana) [reply src/api_request.py:26]: small thing, but why did you change this in the first place? For me it's clearer with geom[0] and geom[1]

Comment (thomasbrockmeier-ams) [reply src/api_request.py:26]: The order in which the coordinates are returned seems to have changed, or is inconsistent. Regardless, projecting the coordinates to RD would throw an error. Considering RD is a projection that is only used in the Netherlands, the larger value in the coordinate pair should always precede the smaller value.

File: src/api_request.py (hunk 2 above)
Comment (thomasbrockmeier-ams) [reply src/api_request.py:53]: Previously every project would implement its own code to fetch the panoramas, which is the case here. The panoramas package is supposed to offer that functionality to (new) projects. I guess I could add it as a dependency here as well. It should be used when this code is ported to the pipeline anyway.

File: scripts/masking.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    Arguments:
+    image_width: int = Panorama image width in pixels
+    image_height: int = Panorama image height in pixels
+    heading: Number = Car heading in degrees
+    mask_degrees: Number = Mask size in degrees. The mask will be applied to the front and back of the image, meaning that a mask
+    of 180 degrees will obscure the entire image.
+
+    Returns:
+    npt.NDArray[(image_height, image_width, 3), bool]
```
Comment (laurenssam) [inline scripts/masking.py:19]: Intuitively, a mask is 2d for me, but not really a problem

**Review by laurenssam — APPROVED** [review]
Super nice, maybe nice to add the functionality to have two images instead of one masked image. This will solve for example the case that a container is split due to the 360 degree image!

**Pull request merged**